mattermost/mattermost-server
# [MM-16146] When converting a bot to a user, set the role to system_user

#### Summary
When a bot is created via the `/api/v4/bots` endpoint, and then converted to a user, via the mattermost CLI, that user will not be able to view any joinable teams once logged into to Mattermost.  This was caused by the converted bot user not having any specified roles. 

To fix this, set the role to `model.SYSTEM_USER_ROLE_ID` (resolves to "system_user").

I question whether there would be a case where we would want any other role setting, other than `model.SYSTEM_USER_ROLE_ID`.  The other options would be to set to as system admin or as previously set in this bug, with no roles.

Note that once a bot is converted to a user, the sysadmin of the mattermost instance has the power to adjust any permissions just as a normally created user. 
#### Testing

1. Create bot with `http://localhost:8065/api/v4/bots` endpoint
2. Payload example `{"username":"api_junkbot10","display_name":"api_junkbot4","description":"test bot"}`
3. Convert bot to user with `mattermost user convert api_junkbot10 --user --email api_junkbot10@fakedomain.com --password password1`
4. login as bot and observe availabe teams to join

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-16146
